cmake/therock_detect_python_versions.cmake

```diff
@@ -0,0 +1,89 @@
+# Function to detect available Python versions on the system
+function(therock_detect_python_versions OUT_VERSIONS)
+  cmake_policy(SET CMP0057 NEW)  # Enable IN_LIST operator
+  set(_python_versions)
+  set(_min_version 8)
+  set(_max_version 13)
+
+  # Try to find each Python version
+  foreach(_minor RANGE ${_min_version} ${_max_version})
+    set(_version "3.${_minor}")
+
+    # Try to find this specific Python version
+    find_program(_python_exe
+      NAMES python${_version} python${_version}.exe
+      PATHS
+        /usr/bin
+        /usr/local/bin
+        /opt/python-${_version}/bin
+        $ENV{HOME}/.pyenv/versions/${_version}*/bin
+        $ENV{HOME}/.local/bin
+        C:/Python${_minor}/
+        C:/Python3${_minor}/
+      NO_DEFAULT_PATH
```

Comment:
This looks brittle. On Windows it will also skip one of the most common install locations - APPDATA. I wonder if this probing could use FindPython instead of `find_program` somehow (without a sticky setting that may conflict with the same Python version found for CMake to use.

Being explicit at the top level and building with just the version that CMake is already using would be safer.